release_bot/releasebot.py

```diff
@@ -261,13 +275,20 @@ def run(self):
                 try:
                     if self.new_release.get('trigger_on_issue') and self.find_open_release_issues():
                         if self.new_release.get('labels') is not None:
-                            self.github.put_labels_on_issue(self.new_pr['issue_number'],
-                                                            self.new_release.get('labels'))
+                            if self.conf.dry_run:
+                                self.logger.info("I would add labels to issue #" +
+                                                 str(self.new_pr['issue_number']))
+                            else:
+                                self.github.put_labels_on_issue(self.new_pr['issue_number'],
+                                                                self.new_release.get('labels'))
                         self.make_release_pull_request()
                 except ReleaseException as exc:
                     self.logger.error(exc)
 
-                self.github.add_comment(self.new_release.get('pr_id'))
+                if self.conf.dry_run:
+                    self.logger.info("I would add a comment to the pull request created.")
+                else:
+                    self.github.add_comment(self.new_release.get('pr_id'))
```

Comment:
I've just tried it and in dry mode I see
`INFO   I would add a comment to the pull request created.`
even no comment is added in normal mode.

When you check the [Github.add_comment()](https://github.com/user-cont/release-bot/blob/master/release_bot/github.py#L173) you can see that the comment is actually added only if `self.comment`. So the `I would add a comment ...` message is misleading.
I suggest adding a `dry_mode` attribute to [Github](https://github.com/user-cont/release-bot/blob/master/release_bot/github.py#L97) and do the
```python
if self.dry_run:
  self.logger.info("I would do something ...")
else:
  self.actually_do_something()
```
in its methods.

Reply:
I don't think there is a need to add the dry_run attribute to `github.py`, because it is already in `configuration.py`, and we can source it from there. 

We can move the tests (for `dry_run` mode) from `release_bot.py` to `github.py`.

I am on it!

Reply:
You're right. I've updated my comment.

Reply:
I have moved some conditions to `github.py`. I have also updated another instance where I was doing a similar mistake. Please have a look at it. :D